pkg/sink/sink.go

```diff
@@ -136,3 +155,64 @@ func createRequestURI(apiVersion, namePlural, namespace string) string {
 	uri = path.Join(uri, namePlural)
 	return uri
 }
+
+func (r Resource) secureEndpoint(trigger triggersv1.Trigger, headers http.Header, payload []byte) error {
+
+	params := []pipelinev1.Param{}
+
+	params = append(params, trigger.TriggerValidate.Params...)
+	params = append(params, pipelinev1.Param{
+		Name: "Payload",
+		Value: pipelinev1.ArrayOrString{
+			Type:      pipelinev1.ParamTypeArray,
+			StringVal: string(payload),
+		},
+	})
+
+	for key := range headers {
```

Comment:
Are we adding each header as a param? If so, won't the validation task need to declare each header it expects as a declared parameter? If not, the task run will fail with a
`invalid input params: didn''t need these params but they were provided
        anyway: [header1]'`

Reply:
I am using params from task and header to check this now.